File: seed/rust-model/trace/src/trace_response.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct TraceResponse {
    #[serde(rename = "submissionId")]
    pub submission_id: SubmissionId,
    #[serde(rename = "lineNumber")]
    pub line_number: i32,
    #[serde(rename = "returnValue")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub return_value: Option<DebugVariableValue>,
    #[serde(rename = "expressionLocation")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub expression_location: Option<ExpressionLocation>,
    pub stack: StackInformation,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub stdout: Option<String>,
}

File: seed/ruby-sdk-v2/trace/lib/seed/submission/types/trace_response.rb
# frozen_string_literal: true

module Seed
  module Submission
    module Types
      class TraceResponse < Internal::Types::Model
        field :submission_id, -> { String }, optional: false, nullable: false, api_name: "submissionId"
        field :line_number, -> { Integer }, optional: false, nullable: false, api_name: "lineNumber"
        field :return_value, lambda {
          Seed::Commons::Types::DebugVariableValue
        }, optional: true, nullable: false, api_name: "returnValue"
        field :expression_location, lambda {
          Seed::Submission::Types::ExpressionLocation
        }, optional: true, nullable: false, api_name: "expressionLocation"
        field :stack, -> { Seed::Submission::Types::StackInformation }, optional: false, nullable: false
        field :stdout, -> { String }, optional: true, nullable: false
      end
    end
  end
end
